src/main/java/com/aws/greengrass/builtin/services/mqttproxy/MqttProxyIPCAgent.java

```diff
@@ -235,9 +236,12 @@ private QualityOfService validateQoS(String qosAsString, String serviceName) {
     }
 
     void doAuthorization(String opName, String serviceName, String topic) throws AuthorizationException {
-        List<String> authorizedResources =
+        if (authorizationHandler.isAuthorized(MQTT_PROXY_SERVICE_NAME,
+                Permission.builder().operation(opName).principal(serviceName).resource(topic).build())) {
+            return;
+        }
+        Set<String> authorizedResources =
```

Comment:
L243+, this should all go away right? Now that we "properly" support MQTT wildcards as part of the auth resource.